src/BlockNode/BlockNode.ts

```diff
@@ -0,0 +1,57 @@
+import { EditorDocument } from '../EditorDocument';
+import { FormattingNode } from '../FormattingNode';
+import { TextNode } from '../TextNode';
+import { ValueNode } from '../ValueNode';
+import { BlockTune } from '../BlockTune';
+import { BlockNodeConstructorParameters } from './types';
+
+/**
+ * BlockNode class represents a node in a tree-like structure used to store and manipulate content in an editor document.
+ * A BlockNode can contain one or more child nodes of type TextNode, DataNode or FormattingNode.
+ * It can also be associated with one or more BlockTunes, which can modify the behavior of the BlockNode.
+ */
+export class BlockNode {
+  /**
+   * Private field representing the name of the BlockNode
+   *
+   * @private
+   */
+  #name: string;
+
+  /**
+   * Private field representing the child nodes of the BlockNode
+   *
+   * @private
+   */
+  #children: Record<string, TextNode | ValueNode | FormattingNode>;
```

Comment:
Let's do like this. So if property is text, it would have array of TextNode | FormattingNode right away

```suggestion
  #children: Record<string, ValueNode | Array<FormattingNode | TextNode>>;
```